hibernate-reactive-core/src/main/java/org/hibernate/reactive/engine/impl/Cascade.java

```diff
@@ -568,16 +568,16 @@ private void cascadeCollectionElements(
 	 */
 	private void deleteOrphans(String entityName, PersistentCollection<?> pc) throws HibernateException {
 		//TODO: suck this logic into the collection!
-		final Collection<?> orphans;
+		final List<?> orphans;
 		if ( pc.wasInitialized() ) {
 			final CollectionEntry ce = eventSource.getPersistenceContextInternal().getCollectionEntry( pc );
 			if ( ce == null ) {
 				return;
 			}
-			orphans = ce.getOrphans( entityName, pc );
+			orphans = Arrays.asList( ce.getOrphans( entityName, pc ).toArray() );
 		}
 		else {
-			orphans = pc.getQueuedOrphans( entityName );
+			orphans = Arrays.asList( pc.getQueuedOrphans( entityName ).toArray() );
```

Comment:
I don't think we want to change the code this way.
You are converting the list to an array and then creating an immutable list  from the array.

Better skip these changes for now and keep the deprecation